### PR TITLE
Feat/bio social links

### PR DESCRIPTION
**Pull request by osmaczko**
### What does the PR do

closes: https://github.com/status-im/status-desktop/issues/5984

See [commits](https://github.com/status-im/status-desktop/pull/6947/commits) for a detailed description.

### Affected areas

- profile settings
- profile view

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

In some places, the design is not matched due to inconsistencies in the design itself, will describe it soon here..

### Videos

https://user-images.githubusercontent.com/33099791/183860776-814ac044-896f-4a4a-80f9-70edcdf21fb2.mp4

https://user-images.githubusercontent.com/33099791/183861133-b4f90f27-8882-45f2-9bc9-0818798d895e.mp4

### CODE REVIEW

File: src/app/modules/shared_models/social_links_model.nim [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      if (self.items[i].uuid == uuid):
+        if (self.items[i].linkType == LinkType.Custom):
+          self.beginRemoveRows(newQModelIndex(), i, i)
+          self.items.delete(i)
+          self.endRemoveRows()
+          return true
+        return false
+    return false
+
+  proc updateItem*(self: SocialLinksModel, uuid, text, url: string): bool =
```
Comment (caybro) [inline src/app/modules/shared_models/social_links_model.nim:62]: What about doing it the right way (tm) by overriding the setData() method? :)

See https://doc.qt.io/qt-5/qabstractitemmodel.html#setData

**Review by caybro — APPROVED** [review]
Looks cool

One suggestion though

**Comment by glitchminer** [comment]
### Issues found 

---
#### 1. OPEN User is required to include whole URL to provide link to social

If a user enters a twitter handle `ethstatus` then should the link on the profile be only to `ethstatus` or to `https://twitter.com/ethstatus`? Please confirm expected behaviour. 

This also applies to the other fields for specific social accounts.

Actual
image
image

---
#### 2. OPEN Max length too long

Custom URLs have a max description of 24 characters but the defined social fields can be long. Decision may be needed on how many chars to show on profile and when clicked, especially for Personal Site (which could be removed in favour of only custom links?)

Actual
image

**Max length of handle** (excluding base URL):
Twitter: 15
Discord: 37 (32+#+4)
Youtube: 20
Github: 39
Telegram: 32

---
#### 3. OPEN Left margin of icon is not per Figma
   Actual
   image

   Expected
   image

---
#### 4. OPEN Custom URLs have a different height than those with icon

Consider increasing height or using same icon as `Personal Site` 

image

---
#### 5. OPEN Not possible to tab between fields

Pressing tab is adding a tab to the input on the field in focus

---
#### 6. OPEN Maximum number of links

Many custom links can be added. Please confirm the expected behaviour of the maximum number of custom links.


---
#### 7. OPEN Order of custom links
The custom links appear in alphabetical order, not the order they are entered in the list.
image

---
#### 7. OPEN Cannot easily clear the 1st custom link

After the first link, it is possible to click on `Remove`. There is not a remove or clear button to easily reset the first custom link.

image

---
#### 8. OBSERVATION ONLY Why use myspace as an example on empty custom links instead of something like `My Status Profile`

---
#### 9. OPEN At min height with many links the remove contact button becomes hidden
image

---
Test info
Build: https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220815-084254-73effb-pr6947.dmg
OS: Mac

**Review by glitchminer — CHANGES_REQUESTED** [review]
Please see comment with issues found: https://github.com/status-im/status-desktop/pull/6947#issuecomment-1216418888

**Review by glitchminer — APPROVED** [review]
Thanks for the fixes @osmaczko! 
Tested and approved with caveat that the items from https://github.com/status-im/status-desktop/pull/6947#issuecomment-1216418888 are discussed in a follow up:

---
#### Item 1: User is required to include whole URL to provide link to social

If a user enters a twitter handle `ethstatus` then should the link on the profile be only to `ethstatus` or to `https://twitter.com/ethstatus`? Please confirm expected behaviour. 
This also applies to the other fields for specific social accounts.

Actual
image
image

---
#### Item 2: Max length too long
While the issue has been resolved it was noted that eliding handles and URLs at max length may leave them obscured until copied
For example, max length of github account is 39, which is longer than both the 16 and 36 char limits.

image

---
#### Item 6: Maximum number of links
Many custom links can be added, upper bound to be determined

image

**Pull request merged**